app/src/main/java/com/ncnf/organization/Organization.java

```diff
@@ -0,0 +1,178 @@
+    private GeoPoint location;
+    private String address;
+    private String phoneNumber;
+    private List<String> adminIds;
+    private List<String>  events;
+
+    public Organization(String name, GeoPoint location, String address, String phoneNumber, String originalOwner) {
+
+        this(UUID.randomUUID(), name, location, address, phoneNumber, new ArrayList<String>() {{ add(originalOwner); }}, new ArrayList<>());
+    }
+
+    public Organization(UUID uuid, String name, GeoPoint location, String address, String phoneNumber, List<String> adminIds, List<String> events) {
+        if(isArrayEmpty(adminIds) || !checkAdminIds(adminIds)){
+            throw new IllegalArgumentException("Organization should have at least one admin created when created");
+        }
+
+        this.uuid = uuid;
+        this.name = name;
+        this.location = location;
+        this.address = address;
+        this.phoneNumber = phoneNumber;
+        this.adminIds = adminIds;
+        this.events = events;
+    }
+
+    public boolean checkAdminIds(List<String> l){
+        for(String s : l){
+            if(isStringEmpty(s)){
+                return false;
+            }
+        }
+        return true;
+    }
+
+    public CompletableFuture<DatabaseResponse> saveToDB(DatabaseService db){
+        if(uuid == null || isArrayEmpty(adminIds)){
+            return CompletableFuture.completedFuture(new DatabaseResponse(false, null, new IllegalStateException()));
+        }
+        Map<String, Object> data = new HashMap<>();
+        data.put(UUID_KEY, uuid.toString());
+        data.put(NAME_KEY, name);
+        data.put(LOCATION_KEY, location);
+        data.put(ADDRESS_KEY, address);
+        data.put(PHONE_NB_KEY, phoneNumber);
+        data.put(ADMIN_KEY, adminIds);
+        data.put(ORGANIZED_EVENTS_KEY, events);
+
+        return db.setDocument(ORGANIZATIONS_COLLECTION_KEY + uuid.toString(), data);
+    }
+
+    public UUID getUuid() {
+        return uuid;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public GeoPoint getLocation() {
+        return location;
+    }
+
+    public String getAddress() {
+        return address;
+    }
+
+    public String getPhoneNumber() {
+        return phoneNumber;
+    }
+
+    public List<String> getAdminIds() {
+        return adminIds;
+    }
+
+    public List<String> getEvents() {
+        return Collections.unmodifiableList(events);
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public void setLocation(GeoPoint location) {
+        this.location = location;
+    }
+
+    public void setAddress(String address) {
+        this.address = address;
+    }
+
+    public void setPhoneNumber(String phoneNumber) {
+        this.phoneNumber = phoneNumber;
+    }
+
+    public void setAdminIds(List<String> adminIds) {
+        this.adminIds = new ArrayList<>(adminIds);
+    }
+
+    public void setEvents(List<String> events) {
```

Comment:
Same comments here. I would suggest creating a method `clearEvents` if you want to have an empty list of events for better clarity